netmap/node_info_test.go

```diff
@@ -294,6 +294,74 @@ func TestNodeInfo_SetContinentName(t *testing.T) {
 	require.Equal(t, anyValidContinentName, n.ContinentName())
 }
 
+func TestNodeInfo_SetAttributes(t *testing.T) {
+	var n netmap.NodeInfo
+	require.Zero(t, n.NumberOfAttributes())
+	n.IterateAttributes(func(string, string) {
+		t.Fatal("handler must not be called")
+	})
+
+	const k1, v1 = "k1", "v1"
+	const k2, v2 = "k2", "v2"
+	const empty = ""
+
+	attr1 := [2]string{empty, v1}
+	attr2 := [2]string{k2, empty}
+
+	require.Panics(t, func() {
+		n.SetAttributes(attr1)
+	})
+
+	require.Panics(t, func() {
+		n.SetAttributes(attr2)
+	})
+
+	attr1 = [2]string{k1, v1}
+	n.SetAttributes(attr1)
+	require.Equal(t, attr1[1], n.Attribute(k1))
+	require.Equal(t, empty, n.Attribute(k2))
+	require.Equal(t, 1, n.NumberOfAttributes())
+
+	attr2 = [2]string{k2, v2}
+	n.SetAttributes(attr2)
+	require.Equal(t, attr2[1], n.Attribute(k2))
+	require.Equal(t, empty, n.Attribute(k1))
+	require.Equal(t, 1, n.NumberOfAttributes())
+
+	n.SetAttributes(attr1, attr2)
+	require.Equal(t, attr1[1], n.Attribute(k1))
+	require.Equal(t, attr2[1], n.Attribute(k2))
+	require.Equal(t, 2, n.NumberOfAttributes())
+
+	n.SetAttributes()
```

Comment:
essentially this is a reset API which code seems a bit strange to me. I'd replace variadic parameter with a slice. At the same time, passing individual items like few lines above look a bit nicer w/o slicing. But i still stand for slice prm, but dont insist

Reply:
i am also for slice, but do not insist